src/optimizer/remove_unused_columns.cpp

```diff
@@ -363,18 +571,39 @@ void RemoveUnusedColumns::RemoveColumnsFromLogicalGet(LogicalGet &get) {
 	// with the "selection vector" that includes filter columns
 	idx_t col_idx = 0;
 	get.projection_ids.clear();
+	vector<idx_t> filtered_original_ids;
+	//! Find matching indices between the proj_sel and the col_sel
 	for (auto proj_sel_idx : proj_sel) {
 		for (; col_idx < col_sel.size(); col_idx++) {
 			if (proj_sel_idx == col_sel[col_idx]) {
-				get.projection_ids.push_back(col_idx);
+				filtered_original_ids.push_back(col_idx);
 				break;
 			}
 		}
 	}
+	auto original_ids = state.MoveOriginalIds();
+	for (auto col : filtered_original_ids) {
+		for (idx_t i = 0; i < original_ids.size(); i++) {
+			if (original_ids[i] == col) {
+				get.projection_ids.push_back(i);
```

Comment:
Maybe add a comment explaining that this is required for pushdown extract, as we can create multiple columns from a single `original_id`